onnx/defs/tensor/defs.cc

```diff
@@ -1026,6 +1052,49 @@ ONNX_OPERATOR_SET_SCHEMA(
                 ->mutable_dim((int)axis)
                 ->set_dim_value(temp);
           }
+        })
+        .PartialDataPropagationFunction([](DataPropagationContext& ctx) {
+          const auto input_data = ctx.getInputData(0);
+          const auto starts = ctx.getInputData(1);
+          const auto ends = ctx.getInputData(2);
+          const auto axes = ctx.getNumInputs() >= 4 ? ctx.getInputData(3) : nullptr;
```

Comment:
I think this could potentially confuse the situation where the input is specified (but data-propagation has no value for it) and the case where the input is not specified (and has a default value). May be better to use ```const bool axes_specified = ctx.getNumInputs() >= 4``` instead of  ```axes == nullptr``` down below.

Reply:
Good catch! Updated. Also, I have applied the same idea to steps.